omnigibson/prims/rigid_prim.py

```diff
@@ -321,8 +320,7 @@ def get_position_orientation(self):
     def set_local_pose(self, position=None, orientation=None):
         # Invalidate kinematic-only object pose caches when new pose is set
         if self.kinematic_only:
-            self._kinematic_world_pose_cache = None
-            self._kinematic_local_pose_cache = None
+            self.clear_kinematic_only_cache()
```

Comment:
now looking back, shouldn't we do `self._kinematic_local_pose_cache = (positions, orientations)`  and `self._kinematic_world_pose_cache = (pos, ori)` in the setters rather than the getters?